### PR TITLE
[service.subtitles.legendasdivx] v0.3.0

### PR DESCRIPTION
**Pull request by highlandr**
Since vfs.rar got fixed by Alwinus, extraction code changed to work accordingly.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by highlandr** [comment]
Needs to be on Leia only ... damn

**Comment by highlandr** [comment]
@enen92 :)

File: service.subtitles.legendasdivx/service.py [hunk 1]
```diff
@@ -448,10 +478,17 @@ def Download(id, filename):
...
+                compressed_file = 'rar://' + quote_plus(local_tmp_file) + '/'
+                log(u"Will try to extract...")
+                xbmc_extract(compressed_file,_newtemp)
+                log(u"Success...")
+            except:
+                xbmc.executebuiltin("XBMC.Extract(%s, %s)" % (local_tmp_file.encode("utf-8"), _newtemp))
+                xbmc.sleep(1000)
+    
             ## IF EXTRACTION FAILS, WHICH HAPPENS SOMETIMES ... BUG?? ... WE WILL BROWSE THE RAR FILE FOR MANUAL EXTRACTION ##
+            log(u"Here...")
```
Comment (enen92) [inline service.subtitles.legendasdivx/service.py:491]: please remove

Comment (enen92) [reply service.subtitles.legendasdivx/service.py:491]: or add a useful log line :)

File: service.subtitles.legendasdivx/service.py [hunk 2]
```diff
@@ -448,10 +478,17 @@ def Download(id, filename):
...
-            xbmc.executebuiltin("XBMC.Extract(%s, %s)" % (local_tmp_file.encode("utf-8"), _newtemp))
-            xbmc.sleep(1000)
-
+            try:
+                compressed_file = 'rar://' + quote_plus(local_tmp_file) + '/'
+                log(u"Will try to extract...")
+                xbmc_extract(compressed_file,_newtemp)
+                log(u"Success...")
+            except:
+                xbmc.executebuiltin("XBMC.Extract(%s, %s)" % (local_tmp_file.encode("utf-8"), _newtemp))
```
Comment (enen92) [inline service.subtitles.legendasdivx/service.py:487]: XBMC. prefixes were removed from builtins in Matrix. I hope the version in matrix doesn't include them, for this one it's fine

File: service.subtitles.legendasdivx/service.py (hunk 1 above)
Comment (highlandr) [reply service.subtitles.legendasdivx/service.py]: deleted :)

File: service.subtitles.legendasdivx/service.py (hunk 2 above)
Comment (highlandr) [reply service.subtitles.legendasdivx/service.py:487]: Need to check that before next release on the matrix branch.

Comment (enen92) [reply service.subtitles.legendasdivx/service.py:487]: Yeah that one slipped between our fingers :) I confirmed you need to change it in matrix

**Pull request merged**